olp-cpp-sdk-core/include/olp/core/network2/NetworkSettings.h

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+#include <olp/core/CoreApi.h>
+#include <olp/core/network2/NetworkProxySettings.h>
+#include <olp/core/network2/NetworkTypes.h>
+
+namespace olp {
+namespace network2 {
+
+/**
+ * @brief This class contains configuration for the network.
+ */
+class CORE_API NetworkSettings final {
```

Comment:
I don't see this class being used. How is it going to be used?
I thought we agreed that it will be part of the NetworkRequest or Send() method.